upgrade to bootstrap / bootswatch v3.3.7

svenevs opened the pull request:
- minor doc fixes
    - I believe I got all 3.3.6 into 3.3.7
    - suggestion to use bs3 over 2
- more specific directions on how to upgrade in future
    - missing documentation of `static/bootstrap-sphinx.css_t` for upgrades
    - much more description of what should go down in `TODO.rst`
        - modified the **epic** `tar` packaging to be more explicit (anybody who has `egrep` aliased can't use the original solution)
- fixed `Makefile` to actually use `python3`
    - added simple option for port override (I'm currently running a mission critical app on `8000` xD)
- fixed `requirements.txt` to allow e.g. `sphinx==1.6.2` or higher
    - `setup.py` has neither of these dependencies?
    - if you add them, you'll need to treat python 2 and 3 separately because of fabric?

#### Notes:

- Candidate to close #149
- Officially the `solar` theme is not in the `v3.3.7` bootswatch tag, see [this issue](https://github.com/thomaspark/bootswatch/issues/643)
    - Just added it manually from the main page
- I don't think you actually want to include the `custom` theme, I'm pretty sure it's either an artifact or a basis for creating your own.
- Changed a few locations referring to the now nonexistent `amelia` theme
 
#### Questions:

1. There is a rather long commented out section in `TODO.rst`.  Might be worth deleting?
2. See changes to `README.rst`, in particular the added `.. warning::`.  The question: I didn't really understand what was going on with #159 because I didn't have any trouble building with `sphinx==1.5.1` or `1.5.5` etc.  Basically, if it's actually a hard requirement then maybe the verbiage is fine, but if not then it's too strong?
3. This is not a "you" problem, but when viewing the source of generated pages I noticed all of the javascript gets loaded at the beginning in the `<meta>` rather than at the bottom.  I know next to nothing about front-end, but those are supposed to be loaded last right (sphinx is doing this, not you)?

Happy to undo / revise any of the included, thanks again for the theme :smile:

ryan-roemer commented on README.rst:
```diff
@@ -228,16 +230,31 @@ configured as above, but with the following code::
+.. warning::
 
-Add a `setup` function in "conf.py" with stylesheet paths added relative to the
+   The Sphinx Bootstrap Theme does not provide legacy Sphinx support, the
```
Is this correct?

My understanding is that the new `@import` CSS sheets put into place after #159 works for both sphinx <= 1.5 and sphinx > 1.5. It's just the "how do I override" instructions that differ...

svenevs commented:
Ok I added the diff you wanted in there, made the `demo_server` target depend on `demo` so it builds everything, and changed the verbiage a little bit with respect to sphinx versions.

Yes, I agree I don't think there are any incompatibilities with respect to older sphinx versions.  However, the reason I am confused is because `app.add_stylesheet` has been around for a **long** time, if you search for `add_stylesheet` in the [changelog](http://www.sphinx-doc.org/en/stable/changes.html) you will see it surface as early as `sphinx==1.0`, where the comment of "supports full URIs" for version 1.1 seems to be the important part -- anything after should accept `app.add_stylesheet` without issue.

AKA I'm kind of leaning toward the original issue not being entirely accurate?  When building locally on say `sphinx==1.5.5`, I had no issues with `app.add_stylesheet`, but I might be missing the bigger picture.

ryan-roemer commented:
@svenevs -- Changes look great. I'm going to merge it and release.

Re: the `add_stylesheet`, the situation evolved the other way -- I wasn't even aware that was an option until finding out the old hacky way of `set css_files =` stopped working. A PR to collapse everything into "just use `add_stylesheet`" and not split on Sphinx versions would definitely be welcome!

ryan-roemer commented:
Released in https://pypi.python.org/pypi/sphinx-bootstrap-theme/0.6.0

ryan-roemer commented:
@svenevs -- I don't usually do this on a first PR, but I'm making you a contributor to this project. Your changes here to not only the core project, but the meta-instructions as well have been fabulous. If you've got free time and desire, I'd love to any maintenance help / review that you wish to do. If not, thanks for all the hard work in this PR!

In the future, just open branches instead of fork PRs, as I can more easily jump in to push minor changes.

Feel free to DM me on twitter https://twitter.com/ryan_roemer if you want to chat out any additional details.

pcav commented:
Thanks! Upgraded seamlessly on our website: http://www.faunalia.eu

svenevs commented:
@ryan-roemer Woot!  Hehehe yes I kind of obsess over documenting documentation systems for some reason.  My plate's more of an overflowing bowl right now, but I do have some thoughts / things I'd like to incorporate in the (hopefully not too distant) future.

@pcav great!!! Thanks for giving feedback that it all works as expected :smile:
